tests/core/anthropic/_utils/test_setup_call.py

```diff
@@ -0,0 +1,139 @@
+    )
+    mock_convert_message_params.assert_called_once_with(
+        mock_base_setup_call.return_value[1]
+    )
+    assert messages == mock_convert_message_params.return_value
+    assert inspect.signature(create) == inspect.signature(Anthropic().messages.create)
+
+    mock_base_setup_call.return_value[1] = [
+        {"role": "system", "content": [{"type": "text", "text": "test"}]}
+    ]
+    _, _, messages, _, call_kwargs = setup_call(
+        model="claude-3-5-sonnet-20240620",
+        client=None,
+        fn=MagicMock(),
+        fn_args={},
+        dynamic_config=None,
+        tools=None,
+        json_mode=False,
+        call_params={"max_tokens": 1000},
+        extract=False,
+    )
+    assert call_kwargs["system"] == [{"type": "text", "text": "test"}]
+
+
+@patch(
+    "mirascope.core.anthropic._utils.setup_call.convert_message_params",
+    new_callable=MagicMock,
+)
+@patch("mirascope.core.anthropic._utils.setup_call._utils", new_callable=MagicMock)
+def test_setup_call_json_mode(
+    mock_utils: MagicMock,
+    mock_convert_message_params: MagicMock,
+    mock_base_setup_call: MagicMock,
+) -> None:
+    """Tests the `setup_call` function with JSON mode."""
+    mock_utils.setup_call = mock_base_setup_call
+    mock_utils.json_mode_content = MagicMock()
+    mock_base_setup_call.return_value[1] = [
+        {"role": "user", "content": [{"type": "text", "text": "test"}]}
+    ]
+    mock_base_setup_call.return_value[-1]["tools"] = MagicMock()
+    mock_convert_message_params.side_effect = lambda x: x
+    _, _, messages, _, call_kwargs = setup_call(
+        model="claude-3-5-sonnet-20240620",
+        client=None,
+        fn=MagicMock(),
+        fn_args={},
+        dynamic_config=None,
+        tools=None,
+        json_mode=True,
+        call_params={"max_tokens": 1000},
+        extract=False,
+    )
+    assert messages[-1]["content"][-1] == {  # type: ignore
+        "type": "text",
+        "text": mock_utils.json_mode_content.return_value,
+    }
+    assert "tools" not in call_kwargs
+
+
+@patch(
+    "mirascope.core.anthropic._utils.setup_call.convert_message_params",
+    new_callable=MagicMock,
+)
+@patch("mirascope.core.anthropic._utils.setup_call._utils", new_callable=MagicMock)
+def test_setup_call_extract(
+    mock_utils: MagicMock,
+    mock_convert_message_params: MagicMock,
+    mock_base_setup_call: MagicMock,
+) -> None:
+    """Tests the `setup_call` function with extraction."""
+    mock_utils.setup_call = mock_base_setup_call
+    _, _, _, tool_types, call_kwargs = setup_call(
+        model="claude-3-5-sonnet-20240620",
+        client=None,
+        fn=MagicMock(),
+        fn_args={},
+        dynamic_config=None,
+        tools=None,
+        json_mode=False,
+        call_params={"max_tokens": 1000},
+        extract=True,
+    )
+    assert "tool_choice" in call_kwargs and call_kwargs["tool_choice"] == {
+        "type": "tool",
+        "name": tool_types[0]._name(),  # type: ignore
```

Comment:
rather than type ignore this, we should first assert that `tool_types[0]` is an instance of `AnthropicTool`

Reply:
done, just asserting its a list has the same effect